### PR TITLE
Homepage graphql queries

### PR DESCRIPTION
**Pull request by weerd**
### What's this PR do?

This pull request further progresses the new Homepage design. It incorporates the GraphQL work in https://github.com/DoSomething/graphql/pull/197 and continues on the styling. There is still a bit more work to be done.

New Home Page progress with Campaigns

### How should this be reviewed?

👀 and maybe some clickity-clicks on the review app.

### Any background context you want to provide?

A couple things to note. The top header is not finalized just yet, and the first "featured" item in the campaign gallery is not completely styled/behaving as it should based on the mocks. More on that to come, but I didn't want the PR to get too complex for review.

### Relevant tickets

References [Pivotal #171056375](https://www.pivotaltracker.com/story/show/171056375).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [x] Added GIF of front-end changes on small, medium, and large screens.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -131,6 +131,7 @@
     "redux-logger": "^3.0.6",
     "redux-thunk": "^2.3.0",
     "sixpack-client": "^2.1.0",
+    "tailwindcss": "^1.2.0",
```
Comment (weerd) [inline package.json]: Moving this into the dependencies list allows us to pull in the `resolveConfig` helper which makes it A LOT easier to dynamically insert Tailwind config values into our JS rendered output when needed!

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import { css } from '@emotion/core';
+import resolveConfig from 'tailwindcss/resolveConfig';
+
+import tailwindConfig from '../../../../../tailwind.config';
+import CampaignGalleryItem from '../../utilities/Gallery/templates/CampaignGalleryItem/CampaignGalleryItemV2';
+import CampaignGalleryFeaturedItem from '../../utilities/Gallery/templates/CampaignGalleryFeaturedItem/CampaignGalleryFeaturedItem';
+
+const HomePageCampaignGallery = ({ campaigns }) => {
+  const tailwindSpacing = resolveConfig(tailwindConfig).theme.spacing;
+  const tailwindScreens = resolveConfig(tailwindConfig).theme.screens;
```
Comment (weerd) [inline resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: This is super helpful. I'll be moving the work I did to separate out the Tailwind Screens values in a separate file, back into the main `tailwind.config.js` file and using this method instead to grab them moving forward!

File: tailwind.config.js [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
+        100: '#f7fafc',
+        200: '#edf2f7',
+        300: '#e2e8f0',
+        400: '#cbd5e0',
+        500: '#a0aec0',
+        600: '#718096',
+        700: '#4a5568',
+        800: '#2d3748',
+        900: '#1a202c',
+      },
```
Comment (weerd) [inline tailwind.config.js]: I was chatting with @lkpttn and we agreed that having the gray values in the config was easier for when needing to look up and reference values in the mocks, as opposed to needing to head into the default Tailwind config in the NPM package. To that end, it made sense to just add the other 3 items we were importing the `colors` variable for and just remove the `require()` statement. Easy peasy.

Donezo

File: resources/assets/components/pages/HomePage/NewHomePage.js [hunk 4]
```diff
@@ -1,15 +1,58 @@
...
+        ... on Showcasable {
+          showcaseTitle
+          showcaseDescription
+          showcaseImage {
+            url
+          }
+        }
+        ... on CampaignWebsite {
+          id
+          url
```
Comment (weerd) [inline resources/assets/components/pages/HomePage/NewHomePage.js]: So I think I need to go back and expose `staffPick` from Contentful to properly do the "Feature" badge on the campaigns within the gallery that aren't the first "featured" item in the gallery. 🤔

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js (hunk 2 above)
Comment (DFurnes) [inline resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: I wonder if we should make a `tailwind.js` that does this for us, so we don't have to import `resolveConfig` and re-process the config files each time we want to access a variable?

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+  return (
+    <ul
+      className="campaign-gallery mt-0"
+      css={css`
+        display: grid;
+        grid-template-columns: 1fr;
+        grid-row-gap: ${tailwindSpacing['8']};
```
Comment (DFurnes) [inline resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: This should also be possible with Tailwind's grid helpers, if you think that reads any nicer:

```js
css={tw`mt-0 grid gap-8 grid-cols-1 md:grid-cols-2 xxl:grid-cols-3`}
```

File: resources/assets/components/pages/HomePage/NewHomePage.js [hunk 6]
```diff
@@ -1,15 +1,58 @@
...
 
-const NewHomePage = () => {
+const HOME_PAGE_QUERY = gql`
+  query HomePageQuery($preview: Boolean!) {
+    page: homePage(preview: $preview) {
+      id
+      title
+      subTitle
+      campaigns {
+        __typename
```
Comment (DFurnes) [inline resources/assets/components/pages/HomePage/NewHomePage.js]: The `__typename` field should be automatically included, I believe.

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js (hunk 2 above)
Comment (weerd) [reply resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: Do you mean maybe a helper function in `/helpers` instead or better as standalone, importable file?

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js (hunk 5 above)
Comment (weerd) [reply resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: Ah nice! I just moved Tailwind package to main dependencies but didn't realize you had also updated to the latest Tailwind version which includes grid classes in your prior PR! I'll review and then update for upcoming PR 💃

File: resources/assets/components/pages/HomePage/HomePageCampaignGallery.js (hunk 2 above)
Comment (DFurnes) [reply resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: Oh, a helper could also be all we need! I'm good either way. 😸

Comment (weerd) [reply resources/assets/components/pages/HomePage/HomePageCampaignGallery.js]: Ok! I'll revise in upcoming PR 💃

**Pull request merged**